vernac/comHints.ml

```diff
@@ -64,18 +64,30 @@ let project_hint ~poly pri l2r r =
   let info = {Typeclasses.hint_priority = pri; hint_pattern = None} in
   (info, true, Hints.hint_globref (GlobRef.ConstRef c))
 
+let warning_deprecated_hint_constr =
+  CWarnings.create_warning ~from:[CWarnings.CoreCategories.automation; CWarnings.CoreCategories.deprecated] ~name:"fragile-hint-constr" ~default:AsError ()
```

Comment:
This could potentially add the category "deprecated since 8.12", but this is definitely nitpicking.

Reply:
For some reason there is no v8_12 subcategory for deprecation... That would require creating a new instance just for this warning.

Reply:
Ok, let's be lazy.

Reply:
It may have been warning since 8.12 but it's only deprecated now so put in in the 8.20 category

Reply:
Done although I think it's not really making sense given that it just raises an error now.